clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/ServiceCachingAgent.java

```diff
@@ -0,0 +1,104 @@
+package com.netflix.spinnaker.clouddriver.ecs.provider.agent;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.ecs.AmazonECS;
+import com.amazonaws.services.ecs.model.DescribeServicesRequest;
+import com.amazonaws.services.ecs.model.ListServicesRequest;
+import com.amazonaws.services.ecs.model.Service;
+import com.netflix.spinnaker.cats.agent.AgentDataType;
+import com.netflix.spinnaker.cats.agent.CacheResult;
+import com.netflix.spinnaker.cats.agent.CachingAgent;
+import com.netflix.spinnaker.cats.agent.DefaultCacheResult;
+import com.netflix.spinnaker.cats.cache.CacheData;
+import com.netflix.spinnaker.cats.cache.DefaultCacheData;
+import com.netflix.spinnaker.cats.provider.ProviderCache;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.ecs.cache.Keys;
+import com.netflix.spinnaker.clouddriver.ecs.provider.EcsProvider;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static com.netflix.spinnaker.cats.agent.AgentDataType.Authority.AUTHORITATIVE;
+import static com.netflix.spinnaker.clouddriver.core.provider.agent.Namespace.CLUSTERS;
+import static com.netflix.spinnaker.clouddriver.ecs.cache.Keys.Namespace.SERVICES;
+
+public class ServiceCachingAgent implements CachingAgent {
+  static final Collection<AgentDataType> types = Collections.unmodifiableCollection(Arrays.asList(
+    AUTHORITATIVE.forType(SERVICES.toString())
+  ));
+
+  private AmazonClientProvider amazonClientProvider;
+  private AWSCredentialsProvider awsCredentialsProvider;
+  private String region;
+  private String accountName;
+
+  public ServiceCachingAgent(String accountName, String region, AmazonClientProvider amazonClientProvider, AWSCredentialsProvider awsCredentialsProvider) {
+    this.accountName = accountName;
+    this.region = region;
+    this.amazonClientProvider = amazonClientProvider;
+    this.awsCredentialsProvider = awsCredentialsProvider;
+  }
+
+  @Override
+  public CacheResult loadData(ProviderCache providerCache) {
+    AmazonECS ecs = amazonClientProvider.getAmazonEcs(accountName, awsCredentialsProvider, region);
+
+    Collection<CacheData> dataPoints = new LinkedList<>();
+    Collection<CacheData> clusters = providerCache.getAll(CLUSTERS.toString());
+
+    for (CacheData cluster : clusters) {
+      if (cluster.getAttributes().get("account") == null || !cluster.getAttributes().get("account").equals(accountName)) {
```

Comment:
As discussed, do remember to remove this block after you're done

Reply:
Done.